### PR TITLE
Adding infinite retry for default.yml fetch

### PR DESCRIPTION
**Pull request by jmervine**
This is a simple attempt at adding a retry to fetching `default.yml` from an http source. Python isn't my strongest language, so I'm happy to address any feedback, or take advice on implementation.

Cheers!

### CODE REVIEW

**Comment by jmervine** [comment]
cc @matthewmodestino

**Comment by jmervine** [comment]
For those kubernetes folks that need this feature now. You can use `initContainers` as a workaround. E.g.
```yaml
      initContainers:
        - name: ensure-defaults
          image: governmentpaas/curl-ssl:latest
          # Requires FQDN, doesn't leverage dnsConfig from above.
          command: [ "sh", "-c", "until curl -f http://defaults-config.splunk.svc.cluster.local/default.yml > /dev/null; do sleep 5; done" ]
        - name: ensure-license
          image: governmentpaas/curl-ssl:latest
          # Requires FQDN, doesn't leverage dnsConfig from above.
          command: [ "sh", "-c", "until curl -f http://defaults-config.splunk.svc.cluster.local/license.lic > /dev/null; do sleep 5; done" ]
      containers:
        - name: license-master
          image: splunk/splunk:latest
          # ...
```

**Comment by jmervine** [comment]
For the Kubernetes based setup I'm working on, I think we'd perfer it to just wait for the default.yml to be available. Perhaps support for a `DEFAULT_HTTP_MAX_TIMEOUT` and `DEFAULT_HTTP_MAX_DELAY` setting, defaulting to 60 and 1200 (seconds) respectively?

**Comment by jmervine** [comment]
@bb03 Let me know what you think of these updates.

File: inventory/environ.py [hunk 1]
```diff
@@ -252,6 +252,9 @@ def push_defaults(url=None):
         if not os.path.exists('/tmp/defaults'):
             os.mkdir('/tmp/defaults')
 
+        max_delay = int(os.environ.get('SPLUNK_DEFAULTS_HTTP_MAX_DELAY', 60))
+        max_timeout = int(os.environ.get('SPLUNK_DEFAULTS_HTTP_MAX_TIMEOUT', 1200))
+        duration = 0
         delay = 1
         while True:
             response = urllib2.urlopen(url)
```
[the remark this answers is not in the document]
Comment (jmervine) [reply inventory/environ.py]: I have no opinion on this. As I said, I'm not a python person. But it sounds reasonable. I don't want to tackle that in this PR though.

[the remark this answers is not in the document]
Comment (lephino) [reply inventory/environ.py]: I think that's fine, just file an issue for it please.  We should move most things around that have the url implementation.  It shores up some code and gets rid of a lot of the complex loop logics.  Makes our lives easy in the long run.

**Comment by lephino** [comment]
Hey @jmervine, I had my team rewrite this with requests and adds in the retry logic.  Take a look at https://github.com/splunk/splunk-ansible/pull/24 and make sure it meets your needs.  Closing this one for that one.

**Pull request closed without merging**